### PR TITLE
Fix retention entity selection

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Previously, target_retention and returning_retention weren't set correctly by the frontend, and then not correctly set in the filter either.

I've massively simplified the frontend logic, and updated the filter to be more correct.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: posthog/models/filters/retention_filter.py [hunk 1]
```diff
@@ -42,9 +44,23 @@ def __init__(self, data: Dict[str, Any] = {}, request: Optional[HttpRequest] = N
         elif not data:
             raise ValueError("You need to define either a data dict or a request")
         self.period = data.get(PERIOD, self.period)
-        self.target_entity = self._parse_target_entity(data.get(TARGET_ENTITY)) or self.target_entity
+        if len(self.entities) > 0:
+            self.target_entity = self.entities[0]
```
Comment (EDsCODE) [inline posthog/models/filters/retention_filter.py:48]: Where does this condition apply now? Before entities array was used on returning entity at most

**Review by EDsCODE — COMMENTED** [review]
other than the comment below looks good!

**Review by EDsCODE — APPROVED** [review]

File: posthog/models/filters/test/test_retention_filter.py [hunk 2]
```diff
@@ -35,11 +56,39 @@ def test_fill_date_from_and_date_to(self):
         self.assertEqual(
             filter.to_dict(),
             {
-                "date_to": "2020-08-02T00:00:00+00:00",
+                "date_to": "2020-08-01",
```
Comment (timgl) [inline posthog/models/filters/test/test_retention_filter.py:59]: @EDsCODE after merging in your "major filter refactoring" I had to change this. Does this still look correct?

File: posthog/models/filters/retention_filter.py [hunk 3]
```diff
@@ -40,7 +40,6 @@ class RetentionFilter(
     PropertyMixin,
     DisplayDerivedMixin,
     InsightMixin,
-    IntervalMixin,
```
Comment (timgl) [inline posthog/models/filters/retention_filter.py:43]: @EDsCODE Also removed IntervalMixin which I think is correct, as we use period here

File: posthog/models/filters/test/test_retention_filter.py (hunk 2 above)
Comment (EDsCODE) [reply posthog/models/filters/test/test_retention_filter.py:62]: yeah I added a comment to describe the difference. i've made it so to_dict will only provide values that have been set which means _date_to and _date_from on the filter object. Before, we were returning derived/default values that date_from/date_to set if there isn't an underlying _date_from/_date_to. The retention filter adds a period to the derived value which is why there's that 1 day difference

File: posthog/models/filters/retention_filter.py (hunk 3 above)
Comment (EDsCODE) [reply posthog/models/filters/retention_filter.py:43]: yup! I also removed some extraneous code i noticed

**Pull request merged**